core/src/main/scala/kafka/server/metadata/BrokerMetadataEventProcessor.scala

```diff
@@ -0,0 +1,156 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server.metadata
+
+import kafka.coordinator.group.GroupCoordinator
+import kafka.coordinator.transaction.TransactionCoordinator
+import kafka.server.{KafkaConfig, MetadataCache, QuotaFactory, ReplicaManager}
+import kafka.utils.Logging
+import org.apache.kafka.common.internals.FatalExitError
+import org.apache.kafka.common.protocol.ApiMessage
+
+/**
+ * Ability to process ApiMessage
+ */
+trait ApiMessageProcessor {
```

Comment:
Based on the numbers we've been kicking around (64 MB max batch size, average around a hundred bytes per message), we could have thousands of entries in each batch.

What if the `ApiMessageProcessor` had a method like `eventTypes()` that returned a `Set[Short]` describing which event types it wanted to see?  Then we could build up a list containing only those event types, and send it to the processor.  I think that would probably be more efficient, although I haven't thought about it that hard...

cc @hachikuji

Reply:
I wonder if the cost of splitting up the list -- in terms of time and space -- would offset any efficiency gained by the processors not having to iterate over larger-than-necessary lists.  I would prefer to keep the code as-is for now and revisit this after we have all of our processors in place and we can do some real benchmarking with real data.

Reply:
ok, fair